drivers/wasapi/audio_driver_wasapi.cpp

```diff
@@ -51,6 +53,7 @@ DEFINE_PROPERTYKEY(PKEY_Device_FriendlyName, 0xa45c254e, 0xdf1c, 0x4efd, 0x80, 0
 const CLSID CLSID_MMDeviceEnumerator = __uuidof(MMDeviceEnumerator);
 const IID IID_IMMDeviceEnumerator = __uuidof(IMMDeviceEnumerator);
 const IID IID_IAudioClient = __uuidof(IAudioClient);
+const IID IID_IAudioClient3 = __uuidof(IAudioClient3);
```

Comment:
[IAudioClient3](https://docs.microsoft.com/en-us/windows/win32/api/audioclient/nn-audioclient-iaudioclient3) is supported on Windows 10 only, interface won't be defined on older SDKs and MinGW and cause will fail to build.

Reply:
Since the PR uses a feature exclusive to Windows 10 I tested on both Windows 10 and Windows 7 with the minimum requirements listed [here](https://docs.godotengine.org/en/stable/development/compiling/compiling_for_windows.html) on fresh Windows installations in a virtual machine to ensure Godot could still run on older systems. For testing on Windows 7 I downloaded Visual Studio Community 2017 (v15.922) and installed it with the default settings which installed the Windows 10 SDK version 1809 (10.0.17763.0). IAudioClient3 is defined in the headers and compiles just fine, but will fail to initialise on Windows 7 (if this happens my PR falls back to IAudioClient at runtime). Is there documentation on the minimum required Windows SDK versions Godot should target?

I did not test with MinGW, as I didn't believe there to be a reason why that wouldn't work. Is it confirmed to not work with MinGW?

Reply:
> Is there documentation on the minimum required Windows SDK versions Godot should target?

Godot aims to support Windows 7 and later. As for the Windows SDK version, I don't know exactly but we'd probably like Godot 4.0 to be buildable with Visual Studio 2017 and 2019 (2015 isn't a strict requirement).

Reply:
> [IAudioClient3](https://docs.microsoft.com/en-us/windows/win32/api/audioclient/nn-audioclient-iaudioclient3) is supported on Windows 10 only, interface won't be defined on older SDKs and MinGW and will fail to build.
@bruvzg was correct about MinGW failing to compile; the audioclient.h distributed with MinGW doesn't define the IAudioClient3 interface. 

With this in mind I made some modifications and have successfully cross-compiled IAudioClient3 support from Ubuntu in addition to MinGW on Windows 7. I tested this build on both Windows 7 and Windows 10 with the intended behaviour on the respective platforms. I'll update the PR with this and the requested change as soon as I can.